nuxeo-features/nuxeo-automation/nuxeo-automation-features/src/test/java/org/nuxeo/ecm/automation/core/test/directory/DirectoryOperationsTest.java

```diff
@@ -109,21 +113,73 @@ public void shouldCreateNewEntries() throws Exception {
         ObjectMapper mapper = new ObjectMapper();
         List<String> createdIds = mapper.readValue(result.getString(), new TypeReference<List<String>>() {
         });
+
         assertEquals(2, createdIds.size());
         assertEquals("newContinent", createdIds.get(0));
         assertEquals("anotherContinent", createdIds.get(1));
 
         try (Session directorySession = directoryService.open("continent")) {
-            DocumentModel entry = directorySession.getEntry("newContinent");
-            assertNotNull(entry);
-            assertEquals("newLabel", entry.getProperty("vocabulary", "label"));
-            entry = directorySession.getEntry("anotherContinent");
-            assertNotNull(entry);
-            assertEquals("anotherLabel", entry.getProperty("vocabulary", "label"));
+            try {
+                assertEntriesIn(directorySession);
+            } finally {
+                // tear down
+                createdIds.forEach(directorySession::deleteEntry);
```

Comment:
Perhaps the test framework does the cleanup before and after all the `@Test`s? I didn't add the tear down at the beginning, but the 2nd test failed to create entries because the entries already existed—they had been created by the 1st test.